App/Sagas/HandleSharePhotoRequest.ts

```diff
@@ -0,0 +1,56 @@
+import { call, put } from 'redux-saga/effects'
+import { ActionType } from 'typesafe-actions'
+import RNFS from 'react-native-fs'
+
+import { uploadFile } from './UploadFile'
+import TextileNode from '../../TextileNode'
+import { SharedImage, AddResult } from '../Models/TextileTypes'
+import TextileNodeActions from '../Redux/TextileNodeRedux'
+import ProcessingImagesActions from '../Redux/ProcessingImagesRedux'
+import UIActions from '../Redux/UIRedux'
+
+export function * handleSharePhotoRequest(action: ActionType<typeof UIActions.sharePhotoRequest>) {
```

Comment:
Step #1. This guy handles a share request by either doing the share from wallet thing, or by adding a photo from the camera roll to IPFS, and triggering an upload.